x-pack/filebeat/input/awss3/metrics.go

```diff
@@ -40,10 +44,23 @@ func (m *inputMetrics) Close() {
 	m.unregister()
 }
 
+func (m *inputMetrics) setSQSMessagesWaiting(count int64) {
+	if count == -1 && m.sqsMessagesWaiting == nil {
```

Comment:
Theoretically if this were to be called by more than one goroutine it could lead to a race condition because it accesses `m.sqsMessagesWaiting` before the `sync.Once.Do()` (e.g. the first caller could be in the middle of a setting the value from inside the sync.Once while a second one is beginning a read).

I think the simplest course of action is to simply remove this whole `if`.

Reply:
My concern there is we will be setting -1 as the metric value even if we never have permission. 

It's either that, or we never set -1 when we lose permissions.

Reply:
> It's either that, or we never set -1 when we lose permissions.

True. I was operating with the assumption that if we had permissions at the start that it will continue to have permissions. If it lost permissions midway then the value would stop updating and be stuck on the last value.

Really given that there is only one goroutine accessing this field it should be safe to do away with the `sync.Once` entirely and not worry about concurrency-safety.